mpt-witness-generator/witness/prepare_witness.go

```diff
@@ -116,7 +115,10 @@ func obtainAccountProofAndConvertToWitness(i int, tMod TrieModification, tModsLe
 	} else if tMod.Type == BalanceChanged {
 		statedb.SetBalance(addr, tMod.Balance)
 	} else if tMod.Type == CodeHashChanged {
-		statedb.SetCode(addr, tMod.CodeHash)
+		statedb.SetCodeHash(addr, tMod.CodeHash)
+		// For cases when the wrong account is obtained by PrefetchAccount:
```

Comment:
How do we know we are in that case?

Reply:
That's very good observation! I still need to handle this (for most cases it's ok, but the problems appears when code hash is set after setting nonce/balance)!

Reply:
Ok! Do you intend to handle it in another issue or this PR?

Reply:
Will be in this PR.